### PR TITLE
add dcgm-exporter

### PR DESCRIPTION
**Pull request by yogiseven**
rebase 36 commits  and  add dcgm-exporter.

### CODE REVIEW

File: metrics/Chart.yaml [hunk 1]
```diff
@@ -32,6 +32,10 @@ dependencies:
   name: grafana
   repository: https://grafana.github.io/helm-charts
   version: 6.43.*
+- condition: dcgmeEporter.enabled
```
Comment (dongjiang1989) [inline metrics/Chart.yaml]: value.yaml add `dcgmeEporter.enabled`. Default false

File: metrics/Chart.yaml [hunk 2]
```diff
@@ -32,6 +32,10 @@ dependencies:
...
+  name: dcgm-exporter
+  repository: https://nvidia.github.io/dcgm-exporter/helm-charts
+  version: 3.1.5
```
Comment (dongjiang1989) [inline metrics/Chart.yaml]: `3.*.*` support?

**Review by dongjiang1989 — COMMENTED** [review]
LGTM

File: metrics/charts/dcgm-exporter/README.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+
+1， 配置使用nvidia runtime
```
Comment (imlense) [inline metrics/charts/dcgm-exporter/README.md:2]: remove REAME to OpsCenter README?

File: metrics/charts/dcgm-exporter/values.yaml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+  #   cpu: 100m
+  #   memory: 128Mi
+  # requests:
+  #   cpu: 100m
+  #   memory: 128Mi
+serviceMonitor:
+  enabled: true
+  interval: 15s
+  honorLabels: false
+  additionalLabels: {}
```
Comment (imlense) [inline metrics/charts/dcgm-exporter/values.yaml:87]: add `release: metrics` in `metrics/values.yaml`

**Review by imlense — COMMENTED** [review]
LGTM

File: metrics/Chart.yaml (hunk 2 above)
Comment (yogiseven) [reply metrics/Chart.yaml]: dcgm latest  version is  3.1.8，   3.1.3+  is supported

File: metrics/charts/dcgm-exporter/README.md (hunk 3 above)
Comment (yogiseven) [reply metrics/charts/dcgm-exporter/README.md:2]: OK，i will change it

File: metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml [hunk 5]
```diff
@@ -0,0 +1,796 @@
...
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "DCGM_FI_DEV_POWER_USAGE{instance=~\"${instance}\", gpu=~\"${gpu}\"}",
+          "interval": "",
+          "legendFormat": "GPU {{gpu}}",
```
Comment (dongjiang1989) [inline metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml]: `legendFormat` maybe has a bug. Please retest yourself

You can changed like ```GPU {{`{{`}}gpu{{`}}`}}```

File: metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml [hunk 6]
```diff
@@ -0,0 +1,796 @@
...
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "DCGM_FI_DEV_SM_CLOCK{instance=~\"${instance}\", gpu=~\"${gpu}\"} * 1000000",
+          "format": "time_series",
+          "interval": "",
+          "intervalFactor": 1,
+          "legendFormat": "GPU {{gpu}}",
```
Comment (dongjiang1989) [inline metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml]: as same question

File: metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml [hunk 7]
```diff
@@ -0,0 +1,796 @@
+{{- /*
+Generated from 'cluster-total' from https://grafana.com/grafana/dashboards/12239-nvidia-dcgm-exporter-dashboard/
+Do not change in-place! In order to change this file first read following link:
+https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack/hack
+*/ -}}
+{{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
+{{- if and (or .Values.grafana.enabled .Values.grafana.forceDeployDashboards) (semverCompare ">=1.14.0-0" $kubeTargetVersion) (semverCompare "<9.9.9-9" $kubeTargetVersion) .Values.grafana.defaultDashboardsEnabled .Values.eBPFExporter.enabled}}
```
Comment (dongjiang1989) [inline metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml]: `.Values.eBPFExporter.enabled`  maybe `.Values.xxxx.enabled``

File: metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml [hunk 8]
```diff
@@ -0,0 +1,796 @@
...
+        "enable": true,
+        "hide": true,
+        "iconColor": "rgba(0, 211, 255, 1)",
+        "name": "Annotations & Alerts",
+        "type": "dashboard"
+      }
+    ]
+  },
+  "description": "This dashboard is to display the metrics from DCGM Exporter on a Kubernetes (1.13+) cluster",
+  "editable": true,
```
Comment (dongjiang1989) [inline metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml]: setting `false`

Comment (dongjiang1989) [inline metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml]: setting `false`

File: metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml [hunk 9]
```diff
@@ -0,0 +1,796 @@
...
+  "gnetId": 12239,
+  "graphTooltip": 0,
+  "id": null,
+  "iteration": 1588401887165,
+  "links": [],
```
Comment (dongjiang1989) [inline metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml]: add `{
          "$$hashKey": "object:2302",
          "asDropdown": true,
          "icon": "external link",
          "tags": [],
          "targetBlank": true,
          "title": "",
          "type": "dashboards"
        }`

File: metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml [hunk 10]
```diff
@@ -0,0 +1,796 @@
...
+      "5m",
+      "15m",
+      "30m",
+      "1h",
+      "2h",
+      "1d"
+    ]
+  },
+  "timezone": "",
+  "title": "NVIDIA DCGM Exporter Dashboard",
```
Comment (imlense) [inline metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml]: change to directory. like:  / Kubernetes / GPU Dashboard(NVIDIA)

File: metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml [hunk 11]
```diff
@@ -0,0 +1,796 @@
...
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "DCGM_FI_DEV_GPU_UTIL{instance=~\"${instance}\", gpu=~\"${gpu}\"}",
+          "interval": "",
+          "legendFormat": "GPU {{gpu}}",
```
Comment (imlense) [inline metrics/templates/grafana/dashboards-1.14/dcgm-exporter-dashboard.yaml]:
```suggestion
          "legendFormat": "GPU {{`{{`}}gpu{{`}}`}}",
```

**Comment by dongjiang1989** [comment]
LGTM

**Pull request merged**